django/db/migrations/state.py

```diff
@@ -360,6 +448,7 @@ def __init__(self, app_label, name, fields, options=None, bases=None, managers=N
         self.app_label = app_label
         self.name = name
         self.fields = fields
+        self.all_fields = {f for _, f in fields}
```

Comment:
So this is meant to store inherited fields as well?

Reply:
It is meant to store "target"/remote fields. Meaningly, `ManyToOneRel` instances.
It is actually only used in one place => https://github.com/django/django/pull/12688/files#diff-a71b173d008afea1acd5318ef509bb7cR486
Else, the test https://github.com/django/django/blob/master/tests/migrations/test_autodetector.py#L1116-L1145 fails because we do not detect the dependency between both models, and therefore the dependency between both apps.
See the error
```
======================================================================
FAIL: test_rename_model_reverse_relation_dependencies (migrations.test_autodetector.AutodetectorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "django/tests/migrations/test_autodetector.py", line 1148, in test_rename_model_reverse_relation_dependencies
    self.assertMigrationDependencies(changes, 'testapp', 0, [('otherapp', '__first__')])
  File "django/tests/migrations/test_autodetector.py", line 502, in assertMigrationDependencies
    self.repr_changes(changes, include_dependencies=True),
AssertionError: Migration dependencies mismatch for testapp.auto_1 (expected [('otherapp', '__first__')]):
  testapp:
    auto_1
      <RenameModel  old_name='EntityA', new_name='RenamedEntityA'>
      Dependencies:
        None
```

Two things:
1. I guess that this is not the right approach. I should be able to somehow leverage the `relations` attribute.
2. I don't know if this logic (of checking relations when iterating through fields) should also be applied in other methods. Like model creation detection. But probably yes, so that is missing from my PR currently